src/internal.rs

```diff
@@ -928,89 +931,158 @@ impl BitDepthDependentType for AlPal {
     type T<BD: BitDepth> = [[[[BD::Pixel; 8]; 3]; 32]; 2]; /* [2 a/l][32 bx/y4][3 plane][8 palette_idx] */
 }
 
-#[derive(Clone, Copy)]
-#[repr(C)]
-pub struct Rav1dTaskContext_scratch_compinter_seg_mask {
+#[derive(Clone, Copy, FromZeroes, FromBytes, AsBytes)]
+#[repr(C, align(64))]
+pub struct ScratchCompinter {
     pub compinter: [[i16; 16384]; 2],
     pub seg_mask: [u8; 16384],
 }
 
-pub struct Lap;
+// Larger of the two between `ScratchCompinter` and `[BD::Pixel; 128 * 32]`.
+const SCRATCH_COMPINTER_SIZE: usize = mem::size_of::<ScratchCompinter>();
 
-impl BitDepthDependentType for Lap {
-    type T<BD: BitDepth> = [BD::Pixel; 128 * 32];
-}
+#[derive(Clone, Copy, FromZeroes, FromBytes, AsBytes)]
+#[repr(C, align(64))]
+pub struct ScratchLapInter([u8; SCRATCH_COMPINTER_SIZE]);
 
-#[derive(Clone, Copy)]
-#[repr(C)]
-pub union Rav1dTaskContext_scratch_lap {
-    pub lap: BitDepthUnion<Lap>,
-    pub c2rust_unnamed: Rav1dTaskContext_scratch_compinter_seg_mask,
+impl ScratchLapInter {
+    pub fn lap_mut<BD: BitDepth>(&mut self) -> &mut [BD::Pixel; 128 * 32] {
+        FromBytes::mut_from_prefix(&mut self.0).unwrap()
+    }
```

Comment:
Can we keep the `AlignN`s on the returned types for all of the returned arrays?  `#[derive(FromZeroes, FromBytes, AsBytes)]` works on the `AlignN` types so that shouldn't be a problem at all.

Reply:
We can't easily do that now that the alignment specifiers are on structs, rather than the data being stored in an `AlignN` struct. We also can't derive `AsBytes` for the `AlignN` structs due to the same generics issues I brought up earlier, and manually implementing it wouldn't be correct as there can be padding bytes. I don't think trying to include `AlignN` in the return type is worth any necessary refactoring, even if we could make it work.

Reply:
We can put the derives on them.  I tried it and it works.  It's because they're effectively `#[repr(transparent)]` since they only have one field.

Reply:
I'm still inclined not to do this for the other reasons I mentioned. I think returning the slices without `Align` wrappers is fine. I'm not sure how you're seeing that the derives on the `AlignN` structs works, when I try adding them I get a compiler error saying it's unsupported on generics that are not `repr(transparent)` or `repr(packed)`, which the `AlignN` structs can't be. I'm also pretty sure `AsBytes` on `AlignN` is incorrect, because e.g. `Align32<[u8; 10]>` would have padding bytes.